psydac/api/ast/linalg.py

```diff
@@ -419,15 +420,22 @@ def _compile_pyccel(self, mod, backend, verbose=False):
 
         from pyccel.epyccel import epyccel
 
-        fmod = epyccel(mod,
-                       accelerators = accelerators,
-                       compiler     = compiler,
-                       fflags       = fflags,
-                       comm         = self.comm,
-                       bcast        = True,
-                       folder       = _PYCCEL_FOLDER,
-                       verbose      = verbose)
-
+        if parallel:
+            fmod = epyccel(mod,
+                           accelerators = accelerators,
+                           compiler     = compiler,
+                           fflags       = fflags,
+                           comm         = self.comm,
+                           bcast        = True,
+                           folder       = _PYCCEL_FOLDER,
+                           verbose      = verbose)
+        else:
+            fmod = epyccel(mod,
+                           accelerators = accelerators,
+                           compiler     = compiler,
+                           fflags       = fflags,
+                           folder       = _PYCCEL_FOLDER,
+                           verbose      = verbose)
```

Comment:
I believe that `bcast = True` is the default. Could you not avoid the code duplication and write simply
```suggestion
        fmod = epyccel(mod,
                       accelerators = accelerators,
                       compiler     = compiler,
                       fflags       = fflags,
                       comm         = self.comm if parallel else None,
                       folder       = _PYCCEL_FOLDER,
                       verbose      = verbose)
```
Also, are you always OK with using the MPI rank 0 in the communicator to accelerate the code? We pass `root = 0` by default.

Reply:
deleted in [0a922eb](https://github.com/pyccel/psydac/pull/248/commits/0a922ebe5711d1e83d5f158155d0b4ca99315bae)